record.go

```diff
@@ -251,14 +251,18 @@ func (r RR) toSRV() (SRV, error) {
 	target := fields[3]
 
 	parts := strings.SplitN(r.Name, ".", 3)
-	if len(parts) < 3 {
-		return SRV{}, fmt.Errorf("name %v does not contain enough fields; expected format: '_service._proto.name'", r.Name)
+	if len(parts) < 2 {
+		return SRV{}, fmt.Errorf("name %v does not contain enough fields; expected format: '_service._proto.name' or '_service._proto'", r.Name)
+	}
+	name := "@"
+	if len(parts) == 3 {
+		name = parts[2]
 	}
```

Comment:
One more edge case is a two part name ending with `.`. For example, `_sip._tcp.`. That's not a valid input I think, but it would parse as `Name: ""`.

Reply:
I think that `_sip._tcp.` falls into the case of “semantically valid, but mostly nonsensical”. There's nothing technically preventing the root zone from hosting an SIP server, although it would be pretty weird if it did. So I think that either allowing or disallowing this would both be reasonable.

Reply:
Let's cross that bridge when we come to it (i.e. someone actually has that problem). Thanks!